dayz.kvp

```diff
@@ -0,0 +1,88 @@
+Meta.DisplayName=DayZ
+Meta.Description=DayZ dedicated server
+Meta.OS=Windows, Linux
+Meta.Arch=x86_64
+Meta.Author=Aperture Development, Greelan, IceOfWraith
+Meta.URL=https://store.steampowered.com/app/221100/
+Meta.DisplayImageSource=steam:223350
+Meta.EndpointURIFormat=steam://connect/{ip}:{GenericModule.App.Ports.$QueryPort}
+Meta.ConfigManifest=dayzconfig.json
+Meta.MetaConfigManifest=dayzmetaconfig.json
+Meta.ConfigRoot=dayz.kvp
+Meta.MinAMPVersion=
+Meta.SpecificDockerImage=
+Meta.DockerRequired=False
+Meta.ContainerPolicy=Supported
+Meta.ContainerPolicyReason=
+Meta.Prerequsites=[]
+Meta.ExtraContainerPackages=[]
+Meta.ConfigReleaseState=NotSpecified
+Meta.NoCommercialUsage=False
+App.DisplayName=DayZ
+App.RootDir=./dayz/
+App.BaseDirectory=./dayz/server/
+App.ExecutableWin=server\DayZServer_x64.exe
+App.ExecutableLinux=server/DayZServer
+App.WorkingDir=server
+App.LinuxCommandLineArgs=
+App.WindowsCommandLineArgs=
+App.CommandLineArgs={{$PlatformArgs}} -profiles=./DayZServerData -bepath={{$FullBaseDir}}/battleye -port={{$GamePort}} -config=serverDZ.cfg -dologs -adminlog -netlog -freezeCheck {{$FormattedArgs}}
+App.UseLinuxIOREDIR=False
+App.AppSettings={}
+App.EnvironmentVariables={"LD_LIBRARY_PATH":"./linux64:%LD_LIBRARY_PATH%","SteamAppId":"{{version}}"}
+App.CommandLineParameterFormat=-{0}={1}
+App.CommandLineParameterDelimiter= 
+App.ExitMethod=String
+App.ExitTimeout=30
+App.ExitString=#shutdown
+App.ExitFile=app_exit.lck
+App.HasWriteableConsole=True
+App.HasReadableConsole=True
+App.SupportsLiveSettingsChanges=False
+App.LiveSettingChangeCommandFormat=set {0} "{1}"
+App.ApplicationIPBinding=0.0.0.0
+App.Ports=[{"Protocol":"UDP","Port":2302,"Offset":0,"Range":1,"Ref":"GamePort","Name":"Game Port","Description":"Main game port used by the server","ChildPorts":null},{"Protocol":"UDP","Port":2305,"Offset":0,"Range":1,"Ref":"QueryPort","Name":"Steam Query Port","Description":"Port used for Steam queries","ChildPorts":null},{"Protocol":"TCP","Port":28015,"Offset":0,"Range":1,"Ref":"RemoteAdminPort","Name":"Application Admin Port","Description":"Port used for RCON","ChildPorts":null}]
```

Comment:
I'm not entirely sure either. Bohemia doesn't mention the Steam port. Although as you know it shouldn't matter. But since this is like Arma the other ports probably do matter. This is where I saw 2302 and 2305.

https://community.bistudio.com/wiki/DayZ:Server_Configuration#Additional_Parameters

Reply:
Lol, on Linux at least, netstat shows 2302, 2304 and 2305 open. No query port open. F***ken Bohemia

I will try it without any ports set at all on the command line or config files and see what it opens by default

Reply:
I feel like 2304 wasn't used now or something. It's really pulling from deep in my memory banks though.

Reply:
In that scenario, it opens the ports that I noted above: 2302, 2304, 27016 (all UDP)

Reply:
Note however that none of those are the RemoteAdminPort, because even with that specified in the metaconfig to write to the BattlEye file (and using the template default of 28015), the RCON server does not appear to be started (the temp file is not created in the battleye dir) and the port is not opened.

Reflecting back on this and what I investigated previously (in the Arma context, maybe also for DayZ), I think 2304 is the BattlEye port, but for BattlEye "traffic", not RCON traffic. So that might be why sometimes you see 2305 mentioned for RCON. Essentially two ports for BattlEye stuff.

If you disabled BattlEye from starting, I suspect 2304 would not be opened.

Reply:
I do know Arma and DayZ were different when it came to ports. We found that out at one point. It's really hilarious reading comments online and how nobody has a clue what ports to open or why.

Reply:
And with Arma didn't you see 2305 and 2306 even get opened or something goofy?

Reply:
I'm thinking we should do 2302-2305 just to be safe and follow the majority of the guides?

Reply:
Arma is different yes, has more ports. But similar principles. Full Arma list here, at least one is "reserved": https://community.bistudio.com/wiki/Arma_3:_Dedicated_Server

Reply:
Up to you but I think 2302-2304 would be enough, plus steamquery and RCON

Reply:
But RCON doesn't seem to start with this template anyway

Reply:
Yeah I'm pretty sure the temp file is only created when a client connects for RCON, no? It's been too long. And since AMP is set to STDIO it'll remain off.

Reply:
Well, I tested that by specifying BattlEyeRCON as the admin method, no change. So pretty sure it is not working

Reply:
Partly because the RCON config file is being populated literally with the ParamFieldName `passwordRCON`. Lol

Reply:
Right, the issue is with the capitalisation in the beserver_x64.cfg. If it is made lower case, then the active RCON file is created. If the RCON port is specified in the config, that port is opened *in addition to* 2304. However, if no port is specified in that file, it only opens 2304 (plus game and query port)

Reply:
OK, sorted it. The RCON port needs to be set and can't be 2304. I tried 2305 and it worked. So did 2404.

So I think the template should have 2302 (gameport), 2304 (battleye port), 2305 (RCON port) and 27016 (steamqueryport).

Reply:
BTW, the active file is created regardless of whether a client connects

Signing off now...